### ISSUE TITLE
How do I import the interface `Tables` from the autogenerated module `knex/types/tables`?

### ISSUE THREAD

**Issue by magnusrodseth** [kristiandupont/kanel#386]
I have successfully generated types from my Postgres database. My `knex-tables.ts` looks like this:

```ts
// @generated
// This file is automatically generated by Kanel. Do not modify manually.

import { Knex } from 'knex';
import type { default as Company, CompanyInitializer, CompanyMutator } from './public/Company';
import type { default as Conditions, ConditionsInitializer, ConditionsMutator } from './public/Conditions';
import type { default as SensorData, SensorDataInitializer, SensorDataMutator } from './public/SensorData';
import type { default as StocksRealTime, StocksRealTimeInitializer, StocksRealTimeMutator } from './public/StocksRealTime';
import type { default as Hyper11Chunk, Hyper11ChunkInitializer, Hyper11ChunkMutator } from './_timescaledb_internal/Hyper11Chunk';
import type { default as Hyper22Chunk, Hyper22ChunkInitializer, Hyper22ChunkMutator } from './_timescaledb_internal/Hyper22Chunk';
import type { default as Hyper23Chunk, Hyper23ChunkInitializer, Hyper23ChunkMutator } from './_timescaledb_internal/Hyper23Chunk';
import type { default as Hyper24Chunk, Hyper24ChunkInitializer, Hyper24ChunkMutator } from './_timescaledb_internal/Hyper24Chunk';
import type { default as Hyper25Chunk, Hyper25ChunkInitializer, Hyper25ChunkMutator } from './_timescaledb_internal/Hyper25Chunk';
import type { default as Hyper36Chunk, Hyper36ChunkInitializer, Hyper36ChunkMutator } from './_timescaledb_internal/Hyper36Chunk';

declare module 'knex/types/tables' {
  interface Tables {
    'company': Knex.CompositeTableType<Company, CompanyInitializer, CompanyMutator>;
    'conditions': Knex.CompositeTableType<Conditions, ConditionsInitializer, ConditionsMutator>;
    'sensor_data': Knex.CompositeTableType<SensorData, SensorDataInitializer, SensorDataMutator>;
    'stocks_real_time': Knex.CompositeTableType<StocksRealTime, StocksRealTimeInitializer, StocksRealTimeMutator>;
    '_hyper_1_1_chunk': Knex.CompositeTableType<Hyper11Chunk, Hyper11ChunkInitializer, Hyper11ChunkMutator>;
    '_hyper_2_2_chunk': Knex.CompositeTableType<Hyper22Chunk, Hyper22ChunkInitializer, Hyper22ChunkMutator>;
    '_hyper_2_3_chunk': Knex.CompositeTableType<Hyper23Chunk, Hyper23ChunkInitializer, Hyper23ChunkMutator>;
    '_hyper_2_4_chunk': Knex.CompositeTableType<Hyper24Chunk, Hyper24ChunkInitializer, Hyper24ChunkMutator>;
    '_hyper_2_5_chunk': Knex.CompositeTableType<Hyper25Chunk, Hyper25ChunkInitializer, Hyper25ChunkMutator>;
    '_hyper_3_6_chunk': Knex.CompositeTableType<Hyper36Chunk, Hyper36ChunkInitializer, Hyper36ChunkMutator>;
  }
}
```

I now want to create my Knex instance based on this module `knex/types/tables` in `knex-tables.ts`. In my `lib/knex.ts`, I cannot seem to import the autogenerated `Tables` interface correctly. I only get the empty `Tables` interface from Knex.

<img width="555" alt="Screenshot 2023-03-15 at 13 39 00" src="https://user-images.githubusercontent.com/59113973/225311149-f7fd8f61-c8e5-400f-b1f4-1d7d4ae75024.png">

<img width="758" alt="Screenshot 2023-03-15 at 13 39 15" src="https://user-images.githubusercontent.com/59113973/225311211-cea4ac2c-0d1a-431b-981c-cc5db27b09c6.png">

Perhaps this is more of a technical TypeScript question: How do I import the interface `Tables` from the autogenerated module `knex/types/tables`?

**Comment by kristiandupont** [comment]
So the whole way that thing works is a bit magic to me. The way I make it work is with a reference comment:
```typescript
/// <reference path="../../../packages/schema/src/models/knex-tables.ts" />
```

..and then, Knex knows what to do by magic.

**Comment by magnusrodseth** [comment]
@kristiandupont In which file? In the file that the Knex instance is initialized?

**Comment by magnusrodseth** [comment]
@kristiandupont Still doesn't work. Do you have a reproducible example that I could have a look at?

**Comment by magnusrodseth** [comment]
Solved. This is my editor playing tricks on me, but in reality it actually works. I was basing my claim that it didn't work on the fact the my `Knex` interface was typed as `Knex<any, any>`. However, when I went to use my `db` instance (the initialized Knex instance), I got the type safety I was looking for. See screenshots for reference.

**Knex typed as `Knex<any, any>`**

<img width="500" alt="Screenshot 2023-03-15 at 20 44 46" src="https://user-images.githubusercontent.com/59113973/225425281-bae8ea6d-b3b9-4cc1-8cc6-94e52b62ef93.png">

**The same Knex instance giving me type safety**

<img width="548" alt="Screenshot 2023-03-15 at 20 45 15" src="https://user-images.githubusercontent.com/59113973/225425393-bc6c775a-1177-4ad1-abed-eedf533564ab.png">

Closing this now, but you might want to update the documentation for `kanel-knex` with this information, @kristiandupont 😊
